app/code/Magento/CatalogWidget/Test/Mftf/Test/StorefrontAddProductToCartThroughAjaxTest.xml

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontAddProductToCartThroughAjaxTest">
+        <annotations>
+            <features value="Product List Widget"/>
+            <stories value="Add product to cart through ajax in Product List widget"/>
+            <title value="Add product through ajax"/>
+            <description value="Add product to cart through ajax in Product List widget on Cms page"/>
+        </annotations>
+        <before>
+            <!-- 1. Create multiple products and assign to a category. -->
+            <createData entity="ApiCategory" stepKey="createCategory"/>
+            <createData entity="ApiSimpleProduct" stepKey="createFirstSimpleProduct">
+                <requiredEntity createDataKey="createCategory"/>
+            </createData>
+            <!-- 2. Login as Admin -->
+            <actionGroup ref="LoginAsAdmin" stepKey="loginToAdminPanel"/>
+            <!-- 3. Go to Home CMS Page in Admin -->
+            <actionGroup ref="AdminOpenCmsPageActionGroup" stepKey="onAdminCmsHomePage">
+                <argument name="page_id" value="2" />
+            </actionGroup>
+            <!-- 4. Add Catalog Products List to CMS Home Page-->
+            <click selector="{{CmsNewPagePageContentSection.header}}" stepKey="clickExpandContentTabForPage"/>
+            <actionGroup ref="AdminInsertWidgetToCmsPageContentActionGroup" stepKey="insertWidgetToCmsPageContentActionGroup">
+                <argument name="widgetType" value="Catalog Products List"/>
+            </actionGroup>
+            <actionGroup ref="AdminClickInsertWidgetActionGroup" stepKey="clickInsertWidgetButton"/>
+            <actionGroup ref="SaveCmsPageActionGroup" stepKey="clickSaveButton"/>
+        </before>
+        <after>
+            <!-- 5. Remove Data -->
+            <deleteData createDataKey="createFirstSimpleProduct" stepKey="deleteFirstSimpleProduct"/>
+            <deleteData createDataKey="createCategory" stepKey="deleteCategory"/>
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logoutFromAdmin"/>
+        </after>
+        <!-- 6. Go to Home Page -->
+        <amOnPage url="{{StorefrontHomePage.url}}" stepKey="goToHomePage"/>
```

Comment:
Please, build the test using action groups only (see [best practices](https://github.com/magento/magento-functional-tests-migration/wiki/Best-practices) for more details). 

So, you need to review the existing action groups, there's a chance that some of your steps are already covered by an action group (visiting home page, for example). If an action group does not exist, you may create it and refer in the test body.